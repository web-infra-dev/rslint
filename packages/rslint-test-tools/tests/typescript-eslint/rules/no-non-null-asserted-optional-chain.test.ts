import { RuleTester, getFixturesRootDir } from '../RuleTester.ts';

const rootPath = getFixturesRootDir();

const ruleTester = new RuleTester({
  languageOptions: {
    parserOptions: {
      project: ['./tsconfig.json'],
      tsconfigRootDir: rootPath,
    },
  },
});

ruleTester.run('no-non-null-asserted-optional-chain', {
  valid: [
    'foo.bar!;',
    'foo.bar!.baz;',
    'foo.bar!.baz();',
    'foo.bar()!;',
    'foo.bar()!();',
    'foo.bar()!.baz;',
    'foo?.bar;',
    'foo?.bar();',
    '(foo?.bar).baz!;',
    '(foo?.bar()).baz!;',
    'foo?.bar!.baz;',
    'foo?.bar!();',
    "foo?.['bar']!.baz;",
  ],
  invalid: [
    {
      code: 'foo?.bar!;',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: 'foo?.bar;',
            },
          ],
        },
      ],
    },
    {
      code: "foo?.['bar']!;",
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: "foo?.['bar'];",
            },
          ],
        },
      ],
    },
    {
      code: 'foo?.bar()!;',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: 'foo?.bar();',
            },
          ],
        },
      ],
    },
    {
      code: 'foo.bar?.()!;',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: 'foo.bar?.();',
            },
          ],
        },
      ],
    },
    {
      code: '(foo?.bar)!.baz',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: '(foo?.bar).baz',
            },
          ],
        },
      ],
    },
    {
      code: '(foo?.bar)!().baz',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: '(foo?.bar)().baz',
            },
          ],
        },
      ],
    },
    {
      code: '(foo?.bar)!',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: '(foo?.bar)',
            },
          ],
        },
      ],
    },
    {
      code: '(foo?.bar)!()',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: '(foo?.bar)()',
            },
          ],
        },
      ],
    },
    {
      code: '(foo?.bar!)',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: '(foo?.bar)',
            },
          ],
        },
      ],
    },
    {
      code: '(foo?.bar!)()',
      errors: [
        {
          messageId: 'noNonNullOptionalChain',
          suggestions: [
            {
              messageId: 'suggestRemovingNonNull',
              output: '(foo?.bar)()',
            },
          ],
        },
      ],
    },
  ],
});