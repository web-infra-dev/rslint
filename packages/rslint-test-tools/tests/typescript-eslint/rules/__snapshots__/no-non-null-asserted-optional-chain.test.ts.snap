// Rstest Snapshot v1

exports[`no-non-null-asserted-optional-chain > invalid 1`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 10,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 2`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 14,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 3`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 12,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 4`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 13,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 5`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 12,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 6`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 12,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 7`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 12,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 8`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 12,
          "line": 1,
        },
        "start": {
          "column": 1,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 9`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 11,
          "line": 1,
        },
        "start": {
          "column": 2,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;

exports[`no-non-null-asserted-optional-chain > invalid 10`] = `
{
  "diagnostics": [
    {
      "filePath": "src/virtual.ts",
      "message": "Optional chain expressions can return undefined by design - using a non-null assertion is unsafe and wrong.",
      "messageId": "noNonNullOptionalChain",
      "range": {
        "end": {
          "column": 11,
          "line": 1,
        },
        "start": {
          "column": 2,
          "line": 1,
        },
      },
      "ruleName": "no-non-null-asserted-optional-chain",
    },
  ],
  "errorCount": 1,
  "fileCount": 1,
  "ruleCount": 1,
}
`;
